src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/AutoInsert/AutoClosingTagOnAutoInsertProvider.cs

```diff
@@ -94,7 +95,7 @@ public bool TryResolveInsertion(
 
         if (closeAngle.Parent is MarkupStartTagSyntax
             {
-                ForwardSlash: null,
+                ForwardSlash: not { Kind: SyntaxKind.ForwardSlash, IsMissing: false },
```

Comment:
For my own knowledge, more than requiring a code change, but is the `IsMissing` check here required? I would have thought a missing token would always be of type `None`, and would also always equal `default(SyntaxToken)`.

Seems there are a few different ways of checking token validity/presence in the tooling changes, and I'm guessing they're all equivalent, just making sure.

(Also note I'm still pretty jetlagged so might be talking nonsense)

Reply:
Thank you, that was very educational